management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java

```diff
@@ -28,104 +33,109 @@ public class ProviderServiceController {
+        log.info("Processing providerService request: {}", providerServiceRequest);
 
+        ProviderService providerService = providerServiceService.createOrUpdateProviderService(providerServiceRequest);
 
-
-    @PutMapping("/{id}")
-    @Operation(summary = "Update a providerService", description = "Update a providerService by id")
-
-    public ResponseEntity<ApiResponse<ProviderService>> updateProviderService(@PathVariable UUID id, @RequestBody ProviderServiceRequest providerService) {
-        log.info("Updating providerService with id: {} and request: {}", id, providerService);
-        try {
-            ProviderService updatedProviderService = providerServiceService.updateProviderService(id, providerService);
-            log.debug("Updated providerService: {}", updatedProviderService);
-            ApiResponse<ProviderService> response = ApiResponse.<ProviderService>builder()
-                    .status(ApiResponseStatus.SUCCESS)
-                    .result(updatedProviderService)
-                    .build();
-            return new ResponseEntity<>(response, HttpStatus.OK);
-        } catch (RuntimeException e) {
-            log.warn("ProviderService not found with id: {}", id);
-            ApiResponse<ProviderService> response = ApiResponse.<ProviderService>builder()
-                    .status(ApiResponseStatus.ERROR)
-                    .errors(Collections.singletonList(new ErrorResponse("Not Found", "ProviderService not found with id " + id)))
-                    .build();
-            return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
-        }
+        log.debug("Processed providerService: {}", providerService);
+        ApiSuccessResponse<ProviderService> response = ApiSuccessResponse.<ProviderService>builder()
+                .status(ApiResponseStatus.SUCCESS)
+                .result(providerService)
+                .build();
+        return new ResponseEntity<>(response, HttpStatus.CREATED);
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<ApiResponse<ProviderService>> getProviderServiceById(@PathVariable UUID id) {
+    @Operation(summary = "Retrieve a provider service by id", description = "Returns a provider service by id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved provider service by id",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ProviderService.class))),
+            @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource"),
+            @ApiResponse(responseCode = "403", description = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(responseCode = "404", description = "The resource you were trying to reach is not found")
+    })
+    public ResponseEntity<ApiSuccessResponse<ProviderService>> getProviderServiceById(@PathVariable UUID id) {
```

Comment:
Move all logics to the service. clean up the controller.
Single Responsibility Principle